PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ["res.currency"]
-    _inherit = ["res.currency.rate"]
-    _inherit = ["res.lang"]

Total Changes: 9

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/spreadsheet/models/ir_http.py
+++ b/addons/spreadsheet/models/ir_http.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class IrHttp(models.AbstractModel):
-    _inherit = ['ir.http']
+    _inherit = 'ir.http'
 
     def session_info(self):
         """

--- a/addons/spreadsheet/models/res_currency.py
+++ b/addons/spreadsheet/models/res_currency.py
@@ -2,7 +2,7 @@ from odoo import api, models
 
 
 class ResCurrency(models.Model):
-    _inherit = ["res.currency"]
+    _inherit = "res.currency"
 
     @api.model
     def get_company_currency_for_spreadsheet(self, company_id=None):

--- a/addons/spreadsheet/models/res_currency_rate.py
+++ b/addons/spreadsheet/models/res_currency_rate.py
@@ -2,7 +2,7 @@ from odoo import api, fields, models
 
 
 class ResCurrencyRate(models.Model):
-    _inherit = ["res.currency.rate"]
+    _inherit = "res.currency.rate"
 
     @api.model
     def _get_rate_for_spreadsheet(self, currency_from_code, currency_to_code, date=None, company_id=None):

--- a/addons/spreadsheet/models/res_lang.py
+++ b/addons/spreadsheet/models/res_lang.py
@@ -9,7 +9,7 @@ from odoo.addons.spreadsheet.utils.formatting import (
 
 
 class ResLang(models.Model):
-    _inherit = ["res.lang"]
+    _inherit = "res.lang"
 
     @api.model
     def get_locales_for_spreadsheet(self):

--- a/addons/spreadsheet/models/spreadsheet_mixin.py
+++ b/addons/spreadsheet/models/spreadsheet_mixin.py
@@ -15,6 +15,7 @@ from odoo.addons.spreadsheet.utils.validate_data import fields_in_spreadsheet, m
 
 
 class SpreadsheetMixin(models.AbstractModel):
+    _name = 'spreadsheet.mixin'
     _description = "Spreadsheet mixin"
     _auto = False
 
